src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -466,11 +467,278 @@ type Moon {
+                            {
+                              ""action"": ""read""
+                            },
+                            {
+                              ""action"": ""update""
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        },
+                        {
+                          ""role"": ""authenticated"",
+                          ""actions"": [
+                            {
+                              ""action"": ""create""
+                            },
+                            {
+                              ""action"": ""read""
+                            },
+                            {
+                              ""action"": ""update""
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        }
+                      ],
+                        ""mappings"": {
+                          """ + entityBackingColumn + @""": """ + entityExposedName + @"""
+                        }
+                      }
+                  }
+              }");
+
+        }
+
+            [TestCleanup]
         public void CleanupAfterEachTest()
         {
             TestHelper.UnsetAllDABEnvironmentVariables();
+            if (File.Exists(HOT_RELOAD_CONFIG_FILE_NAME))
+            {
+                File.Delete(HOT_RELOAD_CONFIG_FILE_NAME);
+            }
+        }
+
+        /// <summary>
+        /// Start DAB with the pre-formed config file, assert that rest and GQL
+        /// requests for the book database object returns the expected results.
+        /// Then hot-reload the config runtime paths both for rest and GQL, and
+        /// again assert that the requests return the expected results against
+        /// the book database object.
+        /// </summary>
+        /// <returns></returns>
+        [TestCategory(MSSQL_ENVIRONMENT)]
+        [TestMethod("Hot-reload runtime paths.")]
+        public async Task HotReloadConfigRuntimePathsEndToEndTest()
+        {
+            // Arrange
+            string restBookContents = $"{{\"value\":{BOOK_DBO_CONTENTS}}}";
+            string restPath = "/restApi";
+            string gQLPath = "/gQLApi";
+
+            string graphQLQueryName = "books";
+            string graphQLQuery = @"{
+                books(first: 100) {
+                    items {
+                        id
+                        title
+                        publisher_id
+                    }
+                }
+            }";
+
+            GenerateConfigFile(connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}");
+
+            TestServer server = new(Program.CreateWebHostBuilder(new string[] { "--ConfigFileName", HOT_RELOAD_CONFIG_FILE_NAME }));    
```

Comment:
When we continue adding more and more hot reload tests, creating a new testserver to execute all dab startup code may be taxing in terms of total execution time spent setting up a test and checking one thing (especially in the pipelines).

Are there other ways to setup the hot reload test fixture such that we startup DAB with original config a single time and then each tests validates the hot reloaded values?

For example:
1. Test Class Setup (runs once for entire hotReloadTestClass): 
    - Load in original config with one entity and spot check that this original config is loaded (perhaps a single get request to check endpoint is wired up.)
    - Modify original config with your hot reloaded contents.
    - Now your test fixture is ready for each individual test.
2. Each test case validates that your new config had the intended effects:
    - Original REST/GraphQL endpoint -> Not Found
    - New REST/GraphQL endpoint -> found
    - (allows you to add more tests: Validate original AuthN/AuthZ config isn't honored and new config is)

Other benefits here are that your individual tests don't need to be involved in:
- new config creation with custom values. 
- hot reloading the config (this would be done in class setup, and tests just check expected hot reloaded state)

Reply:
having one test server created during TestClassSetup may not be appropriate for this particular set of tests. There will be a situation where that may help make testing more efficient. Keep this in mind for future tests and this feedback need not block this bit of coverage being added.

Reply:
I've added a class initialization, which I think is best unless we end up parallelizing things in the future.